internal/config/settings.go

```diff
@@ -0,0 +1,46 @@
+package config
+
+import (
+	"time"
+)
+
+// Settings stores runtime application configuration
+type Settings struct {
+	ServiceAddress string
+	ServicePort    uint64
+
+	EtcdEndpoints  []string
+	EtcdTxnTimeout time.Duration
+
+	LockGroups map[string]uint64
+}
+
+// Parse parses a TOML configuration file and default values
+// into runtime settings.
+func Parse(fpath string) (*Settings, error) {
```

Comment:
While returning a pointer makes the error return statement feel cleaner it doesn't seem like a `nil` return struct can ever be valid and we immediately assume that the returned struct isn't nil in `commonSetup` if an error isn't returned. Seems like changing it to return a non-pointer and just an empty object when an error occurs would work and reduce chances of hitting a nil-pointer dereference in the future.